### ISSUE TITLE
torch.expm1 backward returns zero gradient at -40.0 although the float64 derivative is finite

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#192406]
### 🐛 Describe the bug

`torch.autograd` returns a zero gradient for `torch.expm1` at a `float64` input where the mathematically correct derivative is finite and representable.

The derivative of `expm1(x)` is `exp(x)`. For `x = -40.0`, the true derivative is approximately `4.248354255291589e-18`, which is a normal finite `float64` value. However, PyTorch returns `0.0` for the gradient.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch

torch.set_default_dtype(torch.float64)

x = torch.tensor(-40.0, requires_grad=True)

y = torch.expm1(x)
y.backward()

actual = x.grad
expected = torch.tensor(4.248354255291589e-18, dtype=torch.float64)

print("actual:")
print(actual)
print("expected:")
print(expected)
```

### Actual vs expected result

```text
actual:
tensor(0.)

expected:
tensor(4.2484e-18)
```

The discrepancy occurs in the backward pass. `torch.expm1(-40.0)` itself is close to `-1`, but its derivative should be `exp(-40.0)`, not `0.0`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93

**Comment by soulitzer** [comment]
Since this requires regressing perf / memory, it's not immediately clear whether this is worth it, but open to changing my mind.
